zipkin-thrift/src/main/thrift/com/twitter/zipkin/zipkinCore.thrift

```diff
@@ -204,5 +204,19 @@ struct Span {
   6: list<Annotation> annotations, # all annotations/events that occured, sorted by timestamp
   8: list<BinaryAnnotation> binary_annotations # any binary annotations
   9: optional bool debug = 0       # if true, we DEMAND that this span passes all samplers
+  /**
+   * Microseconds from epoch, usually taken from the first annotation. Ex.
+   * SERVER_RECV.timestamp
+   */
+  10: optional i64 start_ts,
+  /**
+   * Duration of the span in microseconds, used to support queries. This value
+   * is usually derived by subtracting the last annotation from the first. For
+   * example, SERVER_SEND.timestamp - SERVER_RECV.timestamp.
+   *
+   * Note that this should be treated unsigned. i32 implies trace durations are
+   * not longer than 35.79 minutes.
```

Comment:
I don't think that's a reasonable limitation. Just the other day I had a discussion with the data team where they sometimes have spans measured in hours. I would go for i64.